packages/snap-preact-demo/tests/cypress/integration/finder/finder.spec.js

```diff
@@ -0,0 +1,307 @@
+											const valueToSelect = options[0]?.value;
+											expect(valueToSelect).to.exist;
+
+											cy.get(select).select(valueToSelect, { force: true }).should('have.value', valueToSelect);
+
+											cy.snapController(id).then(({ store }) => {
+												expect(store.selections[index].selected).to.equal(valueToSelect);
+											});
+										});
+									});
+							});
+						});
+					});
+
+					it('can change first selection', function () {
+						if (type !== 'hierarchy') this.skip();
+
+						cy.snapController(id).then((controller) => {
+							const targetSelector = controller.targeters[id].targets.map((target) => target.selector).join(',');
+
+							// ensure previous test has a selection
+							const hasSelected = controller.store.selections.filter((selection) => selection.selected).length >= 1;
+							if (!hasSelected) this.skip();
+
+							// get first dropdown that has been selected
+							// hierarchy will enforce selection in order so we can assume this is always the first dropdown
+							const firstSelectedIndex = 0;
+
+							cy.get(targetSelector)
+								.find('select')
+								.eq(firstSelectedIndex)
+								.then((select) => {
+									cy.snapController(id).then(({ store }) => {
+										// value with the most options and is not selected
+										const valueToSelect = store.selections[firstSelectedIndex].data
+											.filter((selection) => !selection.selected)
+											.sort((a, b) => b.count - a.count)[0]?.value;
+										expect(valueToSelect).to.exist;
+
+										cy.get(select).select(valueToSelect, { force: true }).should('have.value', valueToSelect);
+
+										cy.snapController(id).then(({ store }) => {
+											expect(store.selections[firstSelectedIndex].selected).to.equal(valueToSelect);
+
+											// ensure other previously selected dropdowns have been reset
+											store.selections.forEach((selection, index) => {
+												if (index !== firstSelectedIndex) {
+													expect(selection.selected).to.equal('');
+												}
+											});
+										});
+									});
+								});
+						});
+					});
+
+					it('can click the find button', function () {
+						if (type !== 'hierarchy') this.skip();
+						if (!config.selectors?.finder?.findButton) this.skip();
+
+						cy.snapController(id).then((controller) => {
+							const targetSelector = controller.targeters[id].targets.map((target) => target.selector).join(',');
+
+							cy.on('url:changed', (newUrl) => {
+								const expectedUrl = controller.store.selections.filter((selection) => selection.selected)[0].services.urlManager.href;
+								expect(expectedUrl).to.exist;
+								expect(newUrl).to.contain(expectedUrl.substring(1)); //remove first character '?'
+							});
+
+							// click on find button
+							cy.get(targetSelector).find(config.selectors?.finder?.findButton).should('exist').click({ force: true });
```

Comment:
Do you think we should assert that we ended up on the page specified in `config.url` with applied filters?